src/UnitTests.Shared/MockLogger.cs

```diff
@@ -504,6 +504,52 @@ public void AssertLogContains(bool isCaseSensitive, params string[] contains)
             }
         }
 
+        /// <summary>
+        /// Asserts that log file contains a message with a single parameter.
+        /// </summary>
+        /// <param name="message">The message that contains '{0}' substring.</param>
+        /// <param name="parameter">The parameter.</param>
+        public void AssertLogContainsMessageWithParameter(string message, out string parameter)
+        {
+            parameter = "";
+            string placeholder = "{0}";
+
+            string regexPattern = "";
+            if (message.Contains(placeholder))
+            {
+                regexPattern = message.Replace(placeholder, "(.*)");
+            }
+            else
+            {
+                Assert.True(
+                        false,
+                        $"'{message}' was expected to contain a substring '{{0}}'.");
+            }
+
+            lock (_lockObj)
+            {
+                var reader = new StringReader(FullLog);
+                string currentLine = reader.ReadLine();
+
+                while (currentLine != null)
+                {
+                    var match = Regex.Match(currentLine, regexPattern);
```

Comment:
In a test it doesn't super matter, but don't use this `Regex.Match` overload in a loop--it has to create a regex object from `regexPattern` every time, with some overhead. Instead do `new Regex(regexPattern)` outside the loop and `regex.Match(currentLine)` inside the loop.

Reply:
Thanks! Since I removed the message of the location of the compilation file in binlog, I no longer need this method and removed it altogether.